cmd/deploy/deploy.go

```diff
@@ -508,63 +523,41 @@ func (dc *DeployCommand) deploy(ctx context.Context, opts *Options) error {
 			}
 			oktetoLog.SetStage("")
 		}
-		exit <- nil
-	}()
-	shouldExit := false
-	for {
-		select {
-		case <-stopCmds:
-			oktetoLog.Infof("CTRL+C received, starting shutdown sequence")
-			sp := utils.NewSpinner("Shutting down...")
-			sp.Start()
-			defer sp.Stop()
-			dc.Executor.CleanUp(errors.New("interrupt signal received"))
-			return oktetoErrors.ErrIntSig
-		case err := <-exit:
-			if err != nil {
-				return err
-			}
-			shouldExit = true
-		}
-		if shouldExit {
-			break
-		}
-	}
-
-	stopCompose := make(chan os.Signal, 1)
-	signal.Notify(stopCompose, os.Interrupt)
-	exitCompose := make(chan error, 1)
 
-	go func() {
 		if opts.Manifest.Deploy.ComposeSection != nil {
-			oktetoLog.SetStage("Deploying compose")
-			err := dc.deployStack(ctx, opts)
-			exitCompose <- err
-			return
-		}
-		exitCompose <- nil
-	}()
-	shouldExit = false
-	for {
-		select {
-		case <-stopCmds:
-			os.Unsetenv(model.OktetoDisableSpinnerEnvVar)
-			oktetoLog.Infof("CTRL+C received, starting shutdown sequence")
-			return oktetoErrors.ErrIntSig
-		case err := <-exitCompose:
-			if err != nil {
-				return fmt.Errorf("Error deploying compose: %w", err)
+			if err := dc.deployStack(ctx, opts); err != nil {
+				exit <- err
+				return
 			}
-			shouldExit = true
 		}
-		if shouldExit {
-			break
+
+		if opts.Manifest.Deploy.Divert != nil && opts.Manifest.Deploy.Divert.Namespace != opts.Manifest.Namespace {
+			if err := dc.deployDivert(ctx, opts); err != nil {
+				exit <- err
+				return
+			}
+			oktetoLog.Success("Divert from '%s' successfully configured", opts.Manifest.Deploy.Divert.Namespace)
 		}
+
+		exit <- nil
+	}()
+
+	select {
+	case <-stop:
+		oktetoLog.Infof("CTRL+C received, starting shutdown sequence")
+		sp := utils.NewSpinner("Shutting down...")
+		sp.Start()
+		defer sp.Stop()
+		dc.Executor.CleanUp(errors.New("interrupt signal received"))
```

Comment:
same here, we can use the error we already have?

Reply:
updated!